collect_app/src/androidTest/java/org/odk/collect/android/support/async/TrackingCoroutineDispatcher.kt

```diff
@@ -0,0 +1,19 @@
+package org.odk.collect.android.support.async
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Runnable
+import org.odk.collect.android.support.async.AsyncWorkTracker
+import kotlin.coroutines.CoroutineContext
+
+class TrackingCoroutineDispatcher(private val dispatcher: CoroutineDispatcher) : CoroutineDispatcher() {
+    override fun dispatch(
+        context: CoroutineContext,
+        block: Runnable
+    ) {
+        AsyncWorkTracker.startWork()
+        dispatcher.dispatch(context) {
+            block.run()
```

Comment:
What if there is an exception here? Then `AsyncWorkTracker.finishWork()` will not be called.